src/query/datavalues/src/data_value.rs

```diff
@@ -367,6 +368,23 @@ impl Ord for DataValue {
     }
 }
 
+#[allow(clippy::derive_hash_xor_eq)]
+impl Hash for DataValue {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
```

Comment:
This hash seems incorrect to me. Do we need to add `discriminant` for every enum variant?

Reply:
Yes, you are right.